components/encoder/incremental/incremental_encoder.go

```diff
@@ -248,7 +248,7 @@ func (e *Encoder) Start(ctx context.Context, b board.Board) {
 			case <-e.cancelCtx.Done():
 				return
 			case tick = <-ch:
-				if tick.Name == e.A.Name() {
+				if tick.Name == e.encAName {
```

Comment:
This should fix [flaky test](https://viam.atlassian.net/browse/RSDK-7384?focusedCommentId=29339&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-29339). Avoiding calling name each time because it locks the mutex.